stan/math/rev/core/grad.hpp

```diff
@@ -23,7 +23,7 @@ namespace math {
  * <p>This function does not recover any memory from the computation.
  *
  */
-static void grad() {
+static inline void grad() {
```

Comment:
This and the similar change in set_zero_all_adjoints_nested seem like they're different from the rest of the changes here - why do these functions become `inline`?

Reply:
Static functions that aren't used in a given translation unit give an 'unused-function' warning, but this is suppressed if the static function is declared inline: https://sourceware.org/legacy-ml/gdb/2015-02/msg00064.html

These are small enough functions that I think inlining is fine imo